### ISSUE TITLE
[Fedora] Compilation failed: Incompatibilities with DART

### ISSUE THREAD

**Issue by osrf-migration** [gazebosim/gz-physics#23]
**Original report ([archived issue](https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-physics/issues/23)) by Til Hoff (Bitbucket: [turakar](https://bitbucket.org/%7B5b5196a3-571e-4e4c-bed3-6f72e4dabba8%7D/)).**

----------------------------------------
# Description

Ignition cannot be compiled because ign-physics fails.

# Steps to reproduce

* set up Fedora 29 system
* install dependencies

```
sudo dnf install cmake freeglut-devel ffmpeg-devel freeimage-devel gflags-devel glew-devel gts-devel ogre-devel protobuf-devel protobuf-c-devel qt5 tinyxml2-devel tinyxml-devel pkg-config protobuf-compiler python qt5-qtquickcontrols qt5-qtquickcontrols2 qt5-devel qt5-qtdeclarative-devel qt5-qtquickcontrols2-devel ruby rubygem-ronn uuid-devel libzip-devel jsoncpp-devel libcurl-devel libyaml-devel czmq-devel cppzmq-devel sqlite-devel libwebsockets-devel eigen3-devel
```

* compile [DART from source](http://dartsim.github.io/install_dart_on_ubuntu.html)
* compile [ignition from source](https://www.ignitionrobotics.org/docs/latest/install)

**Expected behaviour**

ign-physics compiles successfully.

**Actual behaviour**

Compilation gives errors:

```
--- stderr: ignition-physics1                                                                                        
/workspacepath/ignitionrobotics/ign-physics/dartsim/src/CustomMeshShape.cc: In constructor &#39;ignition::physics::dartsim::CustomMeshShape::CustomMeshShape(const ignition::common::Mesh&amp;, const Vector3d&amp;)&#39;:
/workspacepath/ignitionrobotics/ign-physics/dartsim/src/CustomMeshShape.cc:207:45: error: &#39;ai_real&#39; does not name a type
         mesh-&gt;mVertices[j][k] = static_cast&lt;ai_real&gt;(v[k]);
                                             ^~~~~~~
/workspacepath/ignitionrobotics/ign-physics/dartsim/src/CustomMeshShape.cc:211:44: error: &#39;ai_real&#39; does not name a type
         mesh-&gt;mNormals[j][k] = static_cast&lt;ai_real&gt;(n[k]);
                                            ^~~~~~~
gmake[2]: *** [dartsim/CMakeFiles/ignition-physics1-dartsim-plugin.dir/build.make:76: dartsim/CMakeFiles/ignition-physics1-dartsim-plugin.dir/src/CustomMeshShape.cc.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
/workspacepath/ignitionrobotics/ign-physics/dartsim/src/SDFFeatures.cc: In member function &#39;virtual ignition::physics::Identity ignition::physics::dartsim::SDFFeatures::ConstructSdfCollision(const ignition::physics::Identity&amp;, const sdf::v8::Collision&amp;)&#39;:
/workspacepath/ignitionrobotics/ign-physics/dartsim/src/SDFFeatures.cc:495:15: error: &#39;class dart::dynamics::DynamicsAspect&#39; has no member named &#39;setSecondaryFrictionCoeff&#39;; did you mean &#39;setFrictionCoeff&#39;?
       aspect-&gt;setSecondaryFrictionCoeff(odeFriction-&gt;Get&lt;double&gt;(&#34;mu2&#34;));
               ^~~~~~~~~~~~~~~~~~~~~~~~~
               setFrictionCoeff
/workspacepath/ignitionrobotics/ign-physics/dartsim/src/SDFFeatures.cc:499:15: error: &#39;class dart::dynamics::DynamicsAspect&#39; has no member named &#39;setSlipCompliance&#39;
       aspect-&gt;setSlipCompliance(odeFriction-&gt;Get&lt;double&gt;(&#34;slip1&#34;));
               ^~~~~~~~~~~~~~~~~
/workspacepath/ignitionrobotics/ign-physics/dartsim/src/SDFFeatures.cc:503:15: error: &#39;class dart::dynamics::DynamicsAspect&#39; has no member named &#39;setSecondarySlipCompliance&#39;
       aspect-&gt;setSecondarySlipCompliance(odeFriction-&gt;Get&lt;double&gt;(&#34;slip2&#34;));
               ^~~~~~~~~~~~~~~~~~~~~~~~~~
/workspacepath/ignitionrobotics/ign-physics/dartsim/src/SDFFeatures.cc:508:15: error: &#39;class dart::dynamics::DynamicsAspect&#39; has no member named &#39;setFirstFrictionDirection&#39;
       aspect-&gt;setFirstFrictionDirection(math::eigen3::convert(fdir1));
               ^~~~~~~~~~~~~~~~~~~~~~~~~
gmake[2]: *** [dartsim/CMakeFiles/ignition-physics1-dartsim-plugin.dir/build.make:154: dartsim/CMakeFiles/ignition-physics1-dartsim-plugin.dir/src/SDFFeatures.cc.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2245: dartsim/CMakeFiles/ignition-physics1-dartsim-plugin.dir/all] Error 2
gmake: *** [Makefile:141: all] Error 2
---
Failed   &lt;&lt;&lt; ignition-physics1  [ Exited with code 2 ]
Aborted  &lt;&lt;&lt; ignition-rendering2
```

# Versions

* Fedora 29
* tested DART versions: 6.9.2 \(latest tag, not accepted by ign-physics\), master, commit 45f8e776b7bbcd6b4e4b3ac91bfa45b2a9dfc861 \(bump version to 6.10.0\)
* ign-physics 1.4.0

**Comment by stefanbuettner** [comment]
I also ran into compilation problems with the latest [DART 6.10.1 release](https://github.com/dartsim/dart/tree/v6.10.1) and the latest [ig-physics 4.1.0](https://github.com/ignitionrobotics/ign-physics/tree/ignition-physics4_4.1.0):
```
dartsim/src/SDFFeatures.cc: In member function ‘virtual ignition::physics::Identity ignition::physics::dartsim::SDFFeatures::ConstructSdfCollision(const ignition::physics::Identity&, const sdf::v11::Collision&)’:
dartsim/src/SDFFeatures.cc:821:15: error: ‘class dart::dynamics::DynamicsAspect’ has no member named ‘setSlipCompliance’; did you mean ‘setPrimarySlipCompliance’?
       aspect->setSlipCompliance(odeFriction->Get<double>("slip1"));
               ^~~~~~~~~~~~~~~~~
               setPrimarySlipCompliance
dartsim/src/ShapeFeatures.cc: In member function ‘virtual double ignition::physics::dartsim::ShapeFeatures::GetShapeFrictionPyramidPrimarySlipCompliance(const ignition::physics::Identity&) const’:
dartsim/src/ShapeFeatures.cc:472:18: error: ‘class dart::dynamics::DynamicsAspect’ has no member named ‘getSlipCompliance’; did you mean ‘getPrimarySlipCompliance’?
   return aspect->getSlipCompliance();
                  ^~~~~~~~~~~~~~~~~
                  getPrimarySlipCompliance
dartsim/src/ShapeFeatures.cc: In member function ‘virtual bool ignition::physics::dartsim::ShapeFeatures::SetShapeFrictionPyramidPrimarySlipCompliance(const ignition::physics::Identity&, double)’:
dartsim/src/ShapeFeatures.cc:508:11: error: ‘class dart::dynamics::DynamicsAspect’ has no member named ‘setSlipCompliance’; did you mean ‘setPrimarySlipCompliance’?
   aspect->setSlipCompliance(_value);
           ^~~~~~~~~~~~~~~~~
           setPrimarySlipCompliance
```

**Comment by scpeters** [comment]
> I also ran into compilation problems with the latest [DART 6.10.1 release](https://github.com/dartsim/dart/tree/v6.10.1) and the latest [ig-physics 4.1.0](https://github.com/ignitionrobotics/ign-physics/tree/ignition-physics4_4.1.0):
> 
> ```
> dartsim/src/SDFFeatures.cc: In member function ‘virtual ignition::physics::Identity ignition::physics::dartsim::SDFFeatures::ConstructSdfCollision(const ignition::physics::Identity&, const sdf::v11::Collision&)’:
> dartsim/src/SDFFeatures.cc:821:15: error: ‘class dart::dynamics::DynamicsAspect’ has no member named ‘setSlipCompliance’; did you mean ‘setPrimarySlipCompliance’?
>        aspect->setSlipCompliance(odeFriction->Get<double>("slip1"));
>                ^~~~~~~~~~~~~~~~~
>                setPrimarySlipCompliance
> dartsim/src/ShapeFeatures.cc: In member function ‘virtual double ignition::physics::dartsim::ShapeFeatures::GetShapeFrictionPyramidPrimarySlipCompliance(const ignition::physics::Identity&) const’:
> dartsim/src/ShapeFeatures.cc:472:18: error: ‘class dart::dynamics::DynamicsAspect’ has no member named ‘getSlipCompliance’; did you mean ‘getPrimarySlipCompliance’?
>    return aspect->getSlipCompliance();
>                   ^~~~~~~~~~~~~~~~~
>                   getPrimarySlipCompliance
> dartsim/src/ShapeFeatures.cc: In member function ‘virtual bool ignition::physics::dartsim::ShapeFeatures::SetShapeFrictionPyramidPrimarySlipCompliance(const ignition::physics::Identity&, double)’:
> dartsim/src/ShapeFeatures.cc:508:11: error: ‘class dart::dynamics::DynamicsAspect’ has no member named ‘setSlipCompliance’; did you mean ‘setPrimarySlipCompliance’?
>    aspect->setSlipCompliance(_value);
>            ^~~~~~~~~~~~~~~~~
>            setPrimarySlipCompliance
> ```

this should be fixed by #249
